src/core/expr/fexpr_count_countna.cc

```diff
@@ -0,0 +1,165 @@
+//------------------------------------------------------------------------------
+// Copyright 2023 H2O.ai
+//
+// Permission is hereby granted, free of charge, to any person obtaining a
+// copy of this software and associated documentation files (the "Software"),
+// to deal in the Software without restriction, including without limitation
+// the rights to use, copy, modify, merge, publish, distribute, sublicense,
+// and/or sell copies of the Software, and to permit persons to whom the
+// Software is furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+// FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+// IN THE SOFTWARE.
+//------------------------------------------------------------------------------
+#include "column/const.h"
+#include "column/latent.h"
+#include "column/countna.h"
+#include "column/count_all_rows.h"
+#include "documentation.h"
+#include "expr/fexpr_func.h"
+#include "expr/eval_context.h"
+#include "expr/workframe.h"
+#include "python/xargs.h"
+#include "stype.h"
+namespace dt {
+namespace expr {
+
+template<bool COUNTNA>
+class FExpr_CountNA : public FExpr_Func {
+  private:
+    ptrExpr arg_;
+
+  public:
+    FExpr_CountNA(ptrExpr &&arg)
+      : arg_(std::move(arg)) {}
+
+    std::string repr() const override {
+      std::string out = COUNTNA? "countna" : "count";
+      out += '(';
+      if (arg_->get_expr_kind() != Kind::None) out += arg_->repr();      
+      out += ')';
+      return out;
+    }
+
+
+    Workframe evaluate_n(EvalContext &ctx) const override {
+      Workframe outputs(ctx);
+      Workframe wf = arg_->evaluate_n(ctx);
+      Groupby gby = ctx.get_groupby();
+      // this covers scenarios where
+      // we dont care about the presence or absence of NAs
+      // we just want the total number of rows
+      bool count_all_rows = arg_->get_expr_kind() == Kind::None;
+
+      if (count_all_rows && !COUNTNA) {
```

Comment:
`countna()` could also be called with no `cols`, in such a case we are currently returning `0` for each group. Is this case covered on this PR?

Reply:
is it useful though?

Reply:
I don't know, but may be good for consistency.

Reply:
yea, I dont think it fits in. we are replicating SQL here, where count(*) returns all rows, whereas count(column) counts the non null rows for that column.

Reply:
Well, but if someone already using it, it makes no sense to remove this functionality.

Reply:
I don’t think anyone is using countna(), the use-case could be countna(cols), where cols could become None at some point. My feeling is that returning 0 or None is better in this case, than throwing an error.